Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import CoreCommands
+import TSCUtility
+import TSCBasic
+import PackageModel
+import Foundation
+
+extension SwiftPackageTool {
+    
+    struct Install: AsyncSwiftCommand {
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(abstract: "Offers the ability to install executable targets of the current package.")
+        }
+        
+        @OptionGroup()
+        var globalOptions: GlobalOptions
+        
+        @Option(help: "The name of the executable product to install")
+        var product: String?
+        
+        func run(_ tool: SwiftTool) async throws {
+            let swiftpmBinDir = try tool.fileSystem.getOrCreateSwiftPMInstalledBinariesDirectory()
+            
+            let env = ProcessInfo.processInfo.environment
+            
+            if let path = env.path, !path.contains(swiftpmBinDir.pathString), !globalOptions.logging.quiet {
+                tool.observabilityScope.emit(warning: "PATH doesn't include \(swiftpmBinDir.pathString)! This means you won't be able to access the installed executables by default, and will need to specify the full path.")
+            }
+            
+            let installedPackagesJSONPath = try tool.fileSystem.dotSwiftPM.appending(component: "installedPackages.json")
+            
+            var alreadyExisting = (try? InstalledPackage.registered(registryPath: .init(installedPackagesJSONPath), withFS: tool.fileSystem)) ?? []
+            
+            let workspace = try tool.getActiveWorkspace()
+            let packageRoot = try tool.getPackageRoot()
+
+            let packageGraph = try await workspace.loadRootPackage(at: packageRoot, observabilityScope: tool.observabilityScope)
+            
+            let possibleCanidates = packageGraph.products.filter { $0.type == .executable }
```

Comment:
Let's install only top-level package products, otherwise the existing code collects products from the whole package graph.

```suggestion
            let possibleCanidates = packageGraph.rootPackages
                .flatMap { $0.products }
                .filter { $0.type == .executable }

```